src/neorados/cls/fifo.h

```diff
@@ -564,6 +578,56 @@ class FIFO {
     return init.result.get();
   }
 
+  /// List the visible entries in a FIFO
+  /// once entries are listed via this function they become invisible for
+  /// the duration of visibility_timeout and will be automatically trimmed
+  /// after the retention_period
+  /// Signature(bs::error_code ec, bs::vector<list_entry> entries, bool more)
+  ///
+  /// More is true if entries beyond the last exist.
+  /// The list entries are of the form:
+  /// data - Contents of the entry
+  /// marker - String representing the position of this entry within the FIFO.
+  /// mtime - Time (on the OSD) at which the entry was pushed.
+  template<typename CT>
+  auto list2(int max_entries, //< Maximum number of entries to fetch
```

Comment:
after separated into its own client this will just be called `list` and replace the above call